src/main/java/com/mojang/brigadier/tree/CommandNode.java

```diff
@@ -176,11 +174,11 @@ public int hashCode() {
 
     @Override
     public int compareTo(final CommandNode<S> o) {
-        return ComparisonChain
-            .start()
-            .compareTrueFirst(this instanceof LiteralCommandNode, o instanceof LiteralCommandNode)
-            .compare(getSortedKey(), o.getSortedKey())
-            .result();
+        if (this instanceof LiteralCommandNode && o instanceof LiteralCommandNode) {
+            return getSortedKey().compareTo(o.getSortedKey());
+        }
+
+        return (o instanceof LiteralCommandNode) ? 1 : -1;
```

Comment:
This code returns the same as the actual code with Guava ComparisonChain but I can change it if you prefer.

Reply:
I don't think it does. The ComparisonChain should return `0` if neither are literals (therefore preserving any existing sort), but this will return `-1`.

Reply:
Oh thanks, I have correct that